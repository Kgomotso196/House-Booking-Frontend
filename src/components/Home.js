import React from 'react';

const Home = () => (
  <div>
    <h1>Home</h1>
    {/* Home content */}
  </div>
);

export default Home;
