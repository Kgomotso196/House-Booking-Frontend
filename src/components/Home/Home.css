.home-page {
  z-index: 2;
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  padding: 10px;
}

.welcome-msg {
  position: fixed;
  font-weight: bolder;
  font-size: larger;
}

.house-list {
  position: relative;
  left: 400px;
  display: flex;
  flex-wrap: nowrap;
  overflow-x: auto;
}

.house-card {
  position: relative;
  margin: 150px 100px;
  text-align: center;
}