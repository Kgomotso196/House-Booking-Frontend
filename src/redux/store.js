import { configureStore } from '@reduxjs/toolkit';
import getReservationSlice from './slices/GetReservationSlice';
import getHosesSlice from './slices/GetHosesSlice';
import postReservationSlice from './slices/PostReservationSlice';
import postHousesSlice from './slices/PostHousesSlice';

const store = configureStore({
  reducer: {
    reservations: getReservationSlice,
    houses: getHosesSlice,
    postReservations: postReservationSlice,
    postHouses: postHousesSlice,
  },
});

export default store;
