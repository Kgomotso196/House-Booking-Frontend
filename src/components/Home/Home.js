import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import houseServiceAPI from '../../services/housesService';
import './Home.css';

const Home = () => {
  const dispatch = useDispatch();
  const houses = useSelector((state) => state.houses.houses);
  const isLoading = useSelector((state) => state.houses.isLoading);
  const error = useSelector((state) => state.houses.error);

  useEffect(() => {
    dispatch(houseServiceAPI.fetchHouses());
  }, [dispatch]);

  const [scrollPosition, setScrollPosition] = useState(0);

  const scrollLeft = () => {
    setScrollPosition((prevPosition) => prevPosition + 200);
  };

  const scrollRight = () => {
    setScrollPosition((prevPosition) => prevPosition - 200);
  };

  if (isLoading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error fetching houses:
        {error}
      </div>
    );
  }

  const simplifiedHouses = houses.map((house, index) => ({
    house_name: house.house_name,
    house_image: house.house_image,
    id: house.id || index,
  }));

  return (
    <div className="home-page">
      <p className="welcome-msg">W E L C O ME&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;T O&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;H O U S E&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;R E S E R V A T I O N S</p>
      <button type="button" className="scroll-btn scroll-left" onClick={scrollLeft}>
        &#9665;
      </button>
      <div className="house-list" style={{ overflowX: 'auto', whiteSpace: 'nowrap', marginLeft: `${scrollPosition}px` }}>
        {simplifiedHouses.map((house) => (
          <div className="house-card" key={house.id}>
            <div className="circular-container">
              <img src={house.house_image} alt={house.house_name} />
            </div>
            <p className="house-name">{house.house_name}</p>
            <button type="button" className="details-button">See Details</button>
            <div className="social-group">
              <span className="social-icon"><i className="fa-brands fa-x-twitter" /></span>
              <span className="social-icon"><i className="fa-brands fa-linkedin-in" /></span>
              <span className="social-icon"><i className="fa-brands fa-github" /></span>
            </div>
          </div>
        ))}
      </div>
      <button type="button" className="scroll-btn scroll-right" onClick={scrollRight}>
        &#9655;
      </button>
    </div>
  );
};

export default Home;
