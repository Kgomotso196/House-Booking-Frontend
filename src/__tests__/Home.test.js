/* eslint-env jest */
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import Home from '../components/Home/Home';

describe('Test House Component', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
