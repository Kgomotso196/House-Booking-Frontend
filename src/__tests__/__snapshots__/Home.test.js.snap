// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test House Component it renders correctly 1`] = `
<div
  className="home-page"
>
  <p
    className="welcome-msg"
  >
    WELCOME TO BOOK HOUSE
  </p>
  <button
    className="scroll-btn scroll-left"
    onClick={[Function]}
    type="button"
  >
    ◁
  </button>
  <div
    className="house-list"
    style={
      Object {
        "marginLeft": "0px",
        "overflowX": "auto",
        "whiteSpace": "nowrap",
      }
    }
  />
  <button
    className="scroll-btn scroll-right"
    onClick={[Function]}
    type="button"
  >
    ▷
  </button>
</div>
`;
